tool/planet/agent.go

```diff
@@ -148,6 +148,26 @@ func startLeaderClient(conf *LeaderConfig, errorC chan error) (leaderClient io.C
 		}
 	})
 
+	// Add a callback to watch for changes to the leader key.
+	// If this node becomes the leader, record a LeaderElected event to the
+	// local timeline.
+	client.AddWatchCallback(conf.LeaderKey, conf.Term/3, func(key, prevVal, newVal string) {
+		// recordEventTimeout specifies the max timeout to record an event
+		const recordEventTimeout = 10 * time.Second
+
+		if newVal != conf.PublicIP {
+			return
+		}
+
+		ctx, cancel := context.WithTimeout(context.Background(), recordEventTimeout)
+		defer cancel()
+
+		agent.RecordLocalEvents(ctx, []*pb.TimelineEvent{
+			pb.NewLeaderElected(agent.GetConfig().Clock.Now(),
```

Comment:
How about recording the old leader IP as well?

Reply:
https://github.com/gravitational/planet/pull/597/commits/116acba092c17e97983c1b12ddf7a4b9de347c09